factory/glideFactoryEntryGroup.py

```diff
@@ -578,17 +571,21 @@ def iterate(parent_pid, sleep_time, advertize_rate, glideinDescript, frontendDes
                                 entry.writeStats()
                                 return_dict[entry.name] = entry.getState()
                             except:
-                                entry.log.warning("Error writing stats for entry '%s'" % (entry.name))
-                                entry.log.exception("Error writing stats for entry '%s': " % (entry.name))
+                                entry.log.warning(f"Error writing stats for entry '{entry.name}'")
+                                entry.log.exception(f"Error writing stats for entry '{entry.name}': ")
 
                         try:
-                            os.write(w, pickle.dumps(return_dict))
-                        except:
-                            # Catch and log exceptions if any to avoid
-                            # runaway processes.
-                            entry.log.exception("Error writing pickled state for entry '%s': " % (entry.name))
-                        os.close(w)
+                            try:
+                                os.write(w, pickle.dumps(return_dict))
+                            except:
+                                # Catch and log exceptions if any to avoid
+                                # runaway processes.
+                                entry.log.exception("Error writing pickled state for entry '%s': " % (entry.name))
```

Comment:
Better to change this to f-string?

Reply:
Changed the code around here. Reverted some (is incorrect to actually use the finally if the open is not inside) and changed some. This comment made me notice that this instruction is out of the loop and is incorrect to use entry. This bug was there since 2013.

Reply:
Marco confirmed that the initial version had the incorrect use of finally which was fixed somewhere along the chain of commits coming in through this PR. Resolving this comment now.